platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationContext.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+/**
+ * The migration context is defined as a base interface for all migration contexts.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface MigrationContext {
```

Comment:
❗️  First and foremost, we need this API since it is designed specifically for migration. That being said, there are some similarities between this framework and the Configurator framework. We support an aggregate set of actions and summarize all the results in a report. **The import side of this framework actually sounds identical to what Configurator set out to do in the first place: setup configs, bundles, features, etc given some serialized state.** @coyotesqrl please correct me if I'm wrong. 

Do we want to investigate an ImportMigrationContext powered by Configurator? It seems to fit perfectly but I'm ready for the pushback.

Reply:
I like that @lessarderic - it gives us time to do analysis properly. +1 for postponing.